tests/rules/SyntaxErrorInPreparedStatementMethodRuleTest.php

```diff
@@ -157,6 +161,10 @@ public function testSyntaxErrorInQueryRule(): void
                            ^ (42601).',
                     319,
                 ],
+                [
+                    'Query expects placeholder :freigabe1u1, but it is missing from values given.',
```

Comment:
plase check the example in this file. 
```
$connection->preparedQuery('INSERT into ada(adaid, gesperrt, email,freigabe1u1) values(:adaid, :gesperrt, :email, :freigabe1u1)', ['adaid' => 1, 'gesperrt' => 0, 'email' => 'test@github.com', 0]);
```

I guess this error is not correct, since the last value of the array will be handled by PDO as the last value of the column list.

please also add new test-cases (at the very end of the file) which show that the rule works for writable queries as you expect it (and also cover cases which it should not error about)

Reply:
I think this query is invalid (because it names a parameter `:freigabe1u1` that it does not define), and the new warning is correct. In my testing, I get the following error if I actually try to execute this query:
> PDOException: SQLSTATE[HY093]: Invalid parameter number: parameter was not defined

The write queries that are already part of the test suite seem to cover all the cases I can think of. I'm happy to add more if you feel that some category has been omitted. Please let me know!

Reply:
I see, thx for testing. we can merge after conflicts are resolved then.